modules/ghlint-snakeyaml/src/test/kotlin/net/twisterrob/ghlint/model/SnakePermissionsTest.kt

```diff
@@ -0,0 +1,132 @@
+package net.twisterrob.ghlint.model
+
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import org.intellij.lang.annotations.Language
+import org.junit.jupiter.api.Test
+
+class SnakePermissionsTest {
+
+	@Test fun `job has permissions with correct values`() {
+		val job = loadJob(
+				"""
+				jobs:
+				  test:
+				    permissions:
+				      contents: read
+				      issues: write
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		job.permissions?.contents shouldBe Access.READ
+		job.permissions?.issues shouldBe Access.WRITE
+	}
+
+	@Test fun `workflow has permissions with correct values`() {
+		val workflow = loadWorkflow(
+				"""
+				on:
+				  push:
+				permissions:
+				  pull-requests: write
+				  id-token: read
+				jobs:
+				  test:
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		workflow.permissions?.pullRequests shouldBe Access.WRITE
+		workflow.permissions?.idToken shouldBe Access.READ
+	}
+
+	@Test fun `job with no permissions is null`() {
+		val job = loadJob(
+				"""
+				jobs:
+				  test:
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		job.permissions shouldBe null
+	}
+
+	@Test fun `job with one permission set, remaining are access NONE`() {
+		val job = loadJob(
+				"""
+				jobs:
+				  test:
+				    permissions:
+				      repository-projects: read
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		)
+
+		job.permissions?.repositoryProjects shouldBe Access.READ
+
+		Permission.entries.forEach {
+			if (it != Permission.REPOSITORY_PROJECTS) {
+				job.permissions?.accessOf(it) shouldBe Access.NONE
+			}
+		}
+	}
+
+	@Test fun `asMap produces map representing exactly what is in the yaml`() {
+		val permissions = loadJob(
+				"""
+				jobs:
+				  test:
+				    permissions:
+				      contents: read
+				      issues: write
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		).permissions
+
+		val map = permissions?.asMap()
+		map?.size shouldBe 2
+		map?.get("contents") shouldBe "read"
+		map?.get("issues") shouldBe "write"
+	}
+
+	@Test fun `job has a new permission not modelled by GHLint`() {
```

Comment:
What do we actually want the behaviour to be if a new permission is defined? I was also surprised that this didn't throw a schema error?

Reply:
You're using loadJob which does a direct parsing. You only get schema validation in rule tests, which is an interesting realization, whops. Pushing a method that will help.

Re behavior, I think it should show up in the map. (The schema is unofficial, it'll warn, but it's possible it is outdated; same for ghlint, i.e. we shouldn't prevent users of ghlint from writing rules that operate on the new permissions just because it's not in the model yet.)

Reply:
Test method: https://github.com/TWiStErRob/net.twisterrob.ghlint/pull/296

Swap direct SnakeYaml.load to test framework's `load()`

Reply:
and for the "new permissions" we probably need `loadUnsafe`